trajopt/src/problem_description.cpp

```diff
@@ -601,7 +601,17 @@ void DynamicCartPoseTermInfo::hatch(TrajOptProb& prob)
   else
   {
     tesseract_environment::EnvState::ConstPtr state = prob.GetEnv()->getCurrentState();
-    Eigen::Isometry3d world_to_base = state->transforms.at(prob.GetKin()->getBaseLinkName());
+    Eigen::Isometry3d world_to_base = Eigen::Isometry3d::Identity();
+    try
+    {
+      world_to_base = state->transforms.at(prob.GetKin()->getBaseLinkName());
+    }
+    catch (const std::exception&)
+    {
+      std::stringstream ss;
+      ss << "Failed to find transform for link '" + prob.GetKin()->getBaseLinkName() + "'; using identity transform instead";
+      CONSOLE_BRIDGE_logError(ss.str().c_str());
+    }
```

Comment:
What case did you run into that this was required? The kinematics object is created from the environment, so the base link of the kinematics should be in the state transforms.

Reply:
In the case where someone accidentally provides the name of a link that doesn't exist (which I've done a few times on accident), the program should handle the exception thrown by `.at()` gracefully rather than dying

Reply:
Since this is not recoverable at this state in the code, I think it would be best to propagate the exception. You could also check if the link exist first and if not use the trajopt PRINT_AND_THROW macro to throw an exception. This way if we ever added a gui on top it could propagate all the way up to the GUI where it is recoverable.

Reply:
I changed this to the `PRINT_AND_THROW` as requested, but that has created a circular dependency with this related `tesseract` [PR](https://github.com/ros-industrial-consortium/tesseract/pull/125). If I make this change here, the `tesseract_motion_planners` unit tests provide an empty string to the `target` variable, resulting in an exception being thrown due to inability to find a link with no name.

Should I drop the commit where I implemented this, and open a new PR after it and [125](https://github.com/ros-industrial-consortium/tesseract/pull/125) get merged?

Reply:
It looks like updating the `.travis.yml` to whitelist only `trajopt` packages solves this issue